provider/src/commonMain/kotlin/at/asitplus/crypto/provider/sign/Verifier.kt

```diff
@@ -0,0 +1,97 @@
+package at.asitplus.crypto.provider.sign
+
+import at.asitplus.KmmResult
+import at.asitplus.catching
+import at.asitplus.crypto.datatypes.CryptoPublicKey
+import at.asitplus.crypto.datatypes.CryptoSignature
+import at.asitplus.crypto.datatypes.SignatureAlgorithm
+import at.asitplus.crypto.ecmath.straussShamir
+
+class InvalidSignature(message: String): Throwable(message)
+
+sealed interface Verifier {
+    val signatureAlgorithm: SignatureAlgorithm
+    val publicKey: CryptoPublicKey
+
+    fun verify(data: SignatureInput, sig: CryptoSignature): KmmResult<Unit>
+
+    sealed class EC
+    @Throws(IllegalArgumentException::class)
+    constructor (
+        final override val signatureAlgorithm: SignatureAlgorithm.ECDSA,
+        final override val publicKey: CryptoPublicKey.EC)
+    : Verifier {
+        init {
+            signatureAlgorithm.requiredCurve?.let {
+                require(publicKey.curve == it)
```

Comment:
Can we McGuyer a creation function that returns a KmmResult? not sure myself whether we want this

Reply:
i don't think this is a good idea in the constructor, but i also don't think the constructor will be the "canonical" way to construct verifiers. once we end up with `X509Algorithm.EC.verifierFor(publicKey)` then that can return `KmmResult`

Reply:
Then be sure to add API Docs, referencing `X509Algorithm.EC.verifierFor(publicKey)` once its there.

Reply:
Sure, the constructor will likely end up internal anyway.